README.md

```diff
@@ -1,2 +1,61 @@
-# comp110-worksheet-5
+### comp110-worksheet-5
 Base repository for COMP110 worksheet 5
+
+
+### Question A
+
+The algorithm compares the two lists; it compares all the items and sees if they match. If it finds any matches then it returns "True", if not then it returns "False" and exits the for loop.
+
+
+### Question B
+
+Running through quadratic would be the worst case because it would take longer compared to other complexities. Using the bubble sort method, doubling the items in the list would actually quadruple the time it takes hence O(n<sup>2</sup>). Not only that but going through all the items in the list is mandatory unlike with a binary search.
+
+
+### Question C
+
+Both for loops are still going through the same list. for i goes from 0 to n - 1. n is the length.
+
+
+### Question D
+
+Again both for loops are connected. J is going through the loop of i and therefore the algorithm will run twice as fast as it only needs to check the length of one loop
+
+
+### Question E
+
+The time complexity is not Quadratic. I believe that it is now linear O(n) because the J for loop now feeds into the I for loop which makes the algorithmn twice as fast. It moreso fits linear now because linear is known for being "faster" than quadratic. Because both for loops are now essentially the same; It has a time complexity of "comparing" the length of n to one set of items - O(n)
```

Comment:
It will still be quadratic -- halving the time gives `O(1/2 n^2)`, which is still `O(n^2)` as we ignore leading constants